src/lib/dnssd/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -858,27 +865,30 @@ FullQName AdvertiserMinMdns::GetCommissioningTxtEntries(const CommissionAdvertis
         snprintf(txtCommissioningMode, sizeof(txtCommissioningMode), "CM=%d", static_cast<int>(params.GetCommissioningMode()));
         txtFields[numTxtFields++] = txtCommissioningMode;
 
-        if (params.GetRotatingDeviceId().HasValue())
+        const auto rotatingDeviceId = params.GetRotatingDeviceId();
+        if (rotatingDeviceId.has_value())
         {
-            snprintf(txtRotatingDeviceId, sizeof(txtRotatingDeviceId), "RI=%s", params.GetRotatingDeviceId().Value());
+            snprintf(txtRotatingDeviceId, sizeof(txtRotatingDeviceId), "RI=%s", *rotatingDeviceId);
             txtFields[numTxtFields++] = txtRotatingDeviceId;
         }
 
-        if (params.GetPairingHint().HasValue())
+        const auto pairingHint = params.GetPairingHint();
```

Comment:
I'll stop commenting on this, but fix them all up, please, if the copies are unintended.

Reply:
I think they are intended or at least they were pre-existing, however I updated the code.

The main reason for this was that clang-tidy was complaining of "unchecked optional access" when we did has_value followed by value when the optional is from a method call (since the method call may return different values each time).